File: Practica 2 - Java RMI/GestorBibliotecaService/src/gestor/biblioteca/service/models/handling/BookSearcher.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package gestor.biblioteca.service.models.handling;

import gestor.biblioteca.service.models.TLibro;
import gestor.biblioteca.service.models.TLibro;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Saúl Rodríguez Naranjo
 */
public class BookSearcher implements Searcher<TLibro>
{
    public static final String ISBN_SEARCH_FIELD = "i";
    public static final String TITLE_SEARCH_FIELD = "t";
    public static final String AUTHOR_SEARCH_FIELD = "a";
    public static final String COUNTRY_SEARCH_FIELD = "p";
    public static final String LANGUAGE_SEARCH_FIELD = "d";
    public static final String ALL_SEARCH_FIELD = "*";

    private final String searchField;
    private final String searchText;

    public BookSearcher(String searchField, String searchText)
    {
        this.searchField = searchField;
        this.searchText = searchText;
    }

    @Override
    public List<TLibro> search(List<TLibro> searched)
    {
        List<TLibro> foundBooks = new ArrayList<>();
        
        if(searchField.equalsIgnoreCase(ISBN_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getIsbn().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        else if(searchField.equalsIgnoreCase(TITLE_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getTitulo().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        else if(searchField.equalsIgnoreCase(AUTHOR_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getAutor().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        else if(searchField.equalsIgnoreCase(COUNTRY_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getPais().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        else if(searchField.equalsIgnoreCase(LANGUAGE_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getIdioma().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        else if(searchField.equalsIgnoreCase(ALL_SEARCH_FIELD))
        {
            for (TLibro book : searched)
            {
                if(book.getIsbn().contains(searchText) || book.getTitulo().contains(searchText)
                        || book.getAutor().contains(searchText) || book.getPais().contains(searchText)
                        || book.getIdioma().contains(searchText))
                {
                    foundBooks.add(book);
                }
            }
        }
        
        return foundBooks;
    }

}

File: Practica 3 - C# Net Remoting/GestorBibliotecaService/GestorBibliotecaService/Data/Handling/BookSearcher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GestorBibliotecaService.Data.Handling
{
    public class BookSearcher : Searcher<TLibro>
    {
        public const string ISBN_SEARCH_FIELD = "i";
        public const string TITLE_SEARCH_FIELD = "t";
        public const string AUTHOR_SEARCH_FIELD = "a";
        public const string COUNTRY_SEARCH_FIELD = "p";
        public const string LANGUAGE_SEARCH_FIELD = "d";
        public const string ALL_SEARCH_FIELD = "*";

        private string searchField;
        private string searchText;

        public BookSearcher(string searchField, string searchText)
        {
            this.searchField = searchField;
            this.searchText = searchText;
        }

        public List<TLibro> Search(List<TLibro> searched)
        {
            List<TLibro> foundBooks = new List<TLibro>();

            searchField = searchField.ToLowerInvariant();

            if (searchField == ISBN_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Isbn.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }
            else if (searchField == TITLE_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Titulo.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }
            else if (searchField == AUTHOR_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Autor.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }
            else if (searchField == COUNTRY_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Pais.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }
            else if (searchField == LANGUAGE_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Idioma.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }
            else if (searchField == ALL_SEARCH_FIELD)
            {
                foreach (TLibro book in searched)
                {
                    if (book.Isbn.Contains(searchText) || book.Titulo.Contains(searchText)
                            || book.Autor.Contains(searchText) || book.Pais.Contains(searchText)
                            || book.Idioma.Contains(searchText))
                    {
                        foundBooks.Add(book);
                    }
                }
            }

            return foundBooks;
        }
    }
}
